weblate/trans/views/create.py

```diff
@@ -498,6 +540,10 @@ def get_form(self, form_class=None, empty=False):
             ).order_project()
             form.branch_data = self.branch_data
         elif isinstance(form, ComponentSelectForm):
+            if self.duplicate_existing_component:
+                self.components |= Component.objects.filter(
```

Comment:
This should filter components user has access to avoid the possibility of cloning restricted component.

```suggestion
                self.components |= Component.objects..filter_access(self.request.user).filter(
```